### ISSUE TITLE
Make xaligned_malloc default

### ISSUE THREAD

**Issue by wolfv** [QuantStack/xsimd#76]
I have benchmarked `posix_memalign` and `_mm_malloc`, and they are quite a bit slower for 32 bit alignments than the `xaligned_malloc/free`.

Or we could enable/disable them by some compile time flags?

**Comment by wolfv** [comment]
Benchmarks on my machine:

```
------------------------------------------------------------
Benchmark                     Time           CPU Iterations
------------------------------------------------------------
eigen_alloc/10               25 ns         24 ns   28519773
eigen_alloc/64               39 ns         39 ns   17822213
eigen_alloc/512              41 ns         40 ns   13900576
eigen_alloc/1000             39 ns         39 ns   17369813
xsimd_alloc<32>/10          115 ns        115 ns    5818221
xsimd_alloc<32>/64          114 ns        114 ns    5698507
xsimd_alloc<32>/512         115 ns        115 ns    5848487
xsimd_alloc<32>/1000        115 ns        115 ns    5605429
xsimd_alloc<16>/10           16 ns         16 ns   43721187
xsimd_alloc<16>/64           33 ns         33 ns   21133221
xsimd_alloc<16>/512          34 ns         34 ns   20861147
xsimd_alloc<16>/1000         33 ns         33 ns   20889357
std_alloc/10                 14 ns         14 ns   48237465
std_alloc/64                 32 ns         32 ns   21677690
std_alloc/512                32 ns         32 ns   21488222
std_alloc/1000               32 ns         32 ns   21408455
mm_alloc<16>/10              15 ns         15 ns   44551964
mm_alloc<16>/64              32 ns         32 ns   21091826
mm_alloc<16>/512             33 ns         33 ns   21234516
mm_alloc<16>/1000            33 ns         32 ns   21207604
mm_alloc<32>/10             113 ns        112 ns    5963394
mm_alloc<32>/64             112 ns        112 ns    5945179
mm_alloc<32>/512            113 ns        113 ns    6094024
mm_alloc<32>/1000           113 ns        113 ns    5965317
posix_alloc<16>/10            3 ns          2 ns  280775787
posix_alloc<16>/64          136 ns        136 ns    4905766
posix_alloc<16>/512         140 ns        139 ns    5055340
posix_alloc<16>/1000          2 ns          2 ns  289115191
posix_alloc<32>/10            2 ns          2 ns  284374957
posix_alloc<32>/64          159 ns        159 ns    4309821
posix_alloc<32>/512         143 ns        143 ns    4772409
posix_alloc<32>/1000          3 ns          3 ns  271402673
```

**Comment by JohanMabille** [comment]
Fixed in #77
